blender_addons/io_md5_exporter/io_export_md5.py

```diff
@@ -993,86 +1013,256 @@ def invoke(self, context, event):
+    json.dump(data, outFile, indent = 2)
+
+  return None
+
+# import getpass
+def exportModuleForTerasology(parentDirectory):
+  data = json.loads(moduleData, object_pairs_hook = OrderedDict)
+  print(data)
+  modelName  = getSuggestedModelName()
+  data["id"] = modelName[:1].upper()+modelName[1:]
+  data["displayName"] = modelName
+  data["description"] = "This module contains a single " + modelName + "model that can be spawned with spawnPrefab" + modelName + ". The module has been generated with the Blender addon from the TeraMisc repository"
+  # data["author"] = getpass.getuser()
+  data["version"] = "0.1.1-SNAPSHOT"
+
+  fileName = "modules.txt"
+  with open(os.path.join(parentDirectory, fileName),'w') as outFile:
+    json.dump(data, outFile, indent = 2)
+
+  return None
+
+
+
+from shutil import copy
+def exportTextureForTerasology(assetDirectory, report = print):
+  blendFilePath = bpy.path.abspath("//")
+  modelName = getSuggestedModelName()
+  textureFileName = modelName[:1].upper()+modelName[1:] + "Texture.png"
+  texturePath = os.path.join(blendFilePath,textureFileName)
+  texturePath2 = bpy.path.abspath(bpy.data.images[0].filepath)
+  print("++++++++++++",bpy.path.abspath(bpy.data.images[0].filepath))
+  if not os.path.isfile(texturePath) or not os.path.isfile(texturePath2):
+    report({'ERROR'}, "Texture File must be present in same folder as the .blend file")
+    return
+
+  # assetDirectory = getTargetTerasologyAssetsDirectory(bpy.context)
+  textureDirectory = os.path.join(assetDirectory,"textures")
+  if not os.path.exists(textureDirectory):
+    os.makedirs(textureDirectory)
+
+  if os.path.isfile(texturePath2):
+    copy(texturePath2,textureDirectory)
+  else:
+    copy(texturePath, textureDirectory)
+  return{'FINISHED'}
 
 class ExportPrefabForTerasology(bpy.types.Operator):
   bl_idname = "export.prefab"
   bl_label = "Export a prefab for Terasology"
 
   def invoke(self, context, event):
-    exportPrefabForTerasology()
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportPrefabForTerasology(assetDirectory)
+
+    return{'FINISHED'}
+
+class ExportMaterialForTerasology(bpy.types.Operator):
+  bl_idname = "export.material"
+  bl_label = "Export a material for Terasology"
+
+  def invoke(self, context, event):
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportMaterialForTerasology(assetDirectory)
 
     return{'FINISHED'}
 
+class ExportTextureForTerasology(bpy.types.Operator):
+  bl_idname = "export.texture"
+  bl_label = "Export a texture for Terasology"
 
+  def invoke(self, context, event):
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportTextureForTerasology(assetDirectory, self.report)
+
+    return{'FINISHED'}
+
+from tempfile import TemporaryDirectory
+from shutil import copy
+import shutil
+import tempfile
 import zipfile,os
 class ExportZipFileForTerasology(bpy.types.Operator):
+  '''Create a zip that can be directly used in Terasology'''

```

Comment:
I think it is good that you added a documentation. I think we should also not use the wording of creating a ZIP file. But instead should talk about exporting a ready to use module jar.